File: frontend/src/domain/entities/Product.ts
// Entidade de domínio para Product
export interface Product {
  id: number;
  name: string;
  price: number;
  image: string;
  category: string;
  description: string;
}

// Entidade de domínio para CartItem
export interface CartItem extends Product {
  quantity: number;
}

// Entidade de domínio para Category
export interface Category {
  id: number;
  name: string;
} 

File: backend/internal/domain/entities/product.go
package entities

import "time"

// Product representa a entidade de domínio de um produto
type Product struct {
	ID          uint      `json:"id"`
	Name        string    `json:"name"`
	Image       string    `json:"image"`
	Price       float64   `json:"price"`
	CategoryID  uint      `json:"category_id"`
	Category    Category  `json:"category"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

// Category representa a entidade de domínio de uma categoria
type Category struct {
	ID        uint      `json:"id"`
	Name      string    `json:"name"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
